### ISSUE TITLE
instance of <subclass> has no <class attribute> member when <class attribute> is first defined with type hint

### ISSUE THREAD

**Issue by radkujawa** [PyCQA/pylint#4219]
### Steps to reproduce

Given file `a.py`:
```
class A:
    CLASS_VAR: int

    def foo1(self):
        print(self.CLASS_VAR)


class B(A):
    CLASS_VAR = 1

    def foo2(self):
        print(self.CLASS_VAR)
```

### Current behavior
```
$ pylint a.py
************* Module file
a.py:12: [E1101(no-member), B.foo2] Instance of 'B' has no 'CLASS_VAR' member

-----------------------------------
Your code has been rated at 3.75/10

```

### Expected behavior
`B` is a class that really really has `CLASS_VAR` member.

### pylint --version output
```
pylint 2.7.2
astroid 2.5.1
Python 3.7.9 (default, Sep 25 2020, 07:46:23) 
[GCC 8.4.0]
```

**Comment by radkujawa** [comment]
Remarks:
* this might be related to https://github.com/PyCQA/pylint/issues/3131 or https://github.com/PyCQA/pylint/issues/3129, as those two issues look similar, both were created for pylint version 2.4.0 and https://github.com/PyCQA/pylint/issues/4219 also is present since 2.4.0. However, this might be also a separate issue.
* Interesting thing is that pylint complaints to `foo2`, but is OK with `foo1`, so this is certainly a bug, not a feature.
* I'm not able to upgrade pylint, because any version >= 2.4.0 is not usable because of many many many false positive errors "Instance of X has no Y member". I'm using 2.3.1, which is prehistoric, as it is more than two years old.

**Comment by cdce8p** [comment]
This is a duplicate of #3167. The fix is already merged and will be included in the next release.
As for #3131 and #3129, they are unrelated issues that still need to be addressed.

**Comment by radkujawa** [comment]
thank you very much for quick response! It's indeed fixed on master branch, I should have checked that first. Sorry!
